### PR TITLE
Improve email address validation

### PR DESCRIPTION
**Pull request by issyl0**
- We were previously only validating that the email address contains the
  `@` character. That's not enough, as people can enter invalid email
  addresses, like ones with commas in, if they mistype `.com` or
  somesuch.
- Our downstream data consumers were finding this out the hard way when
  they tried to use the email addresses to contact people.
- This uses the Python implementation at https://emailregex.com/ to
  validate the email address doesn't contain any invalid characters.

https://trello.com/c/hGSi37CH/315-email-address-field-accepts-commas

### CODE REVIEW

**Review by bilbof — CHANGES_REQUESTED** [review]
If possible could we have some more tests of this helper, with various email combinations? Just want to make sure we have various examples of invalid/valid emails in the test cases.

**Review by bilbof — APPROVED** [review]
🚀 awesome, thank you so much

**Comment by issyl0** [comment]
Thanks for the review, Bill. We can't merge this until we've heard back from DWP.

**Review by leenagupte — DISMISSED** [review]
Needs stakeholder approval.

**Comment by huwd** [comment]
@issyl0 [Steve has confirmed ](https://trello.com/c/hGSi37CH/315-email-address-field-accepts-commas)this can be merged without waiting for a reply

So looks like we can resolve that conflict and :rocket:

**Pull request merged**
